tests/pytests/integration/cli/test_salt_call.py

```diff
@@ -118,16 +119,22 @@ def test_issue_6973_state_highstate_exit_code(salt_call_cli):
     for this minion, salt-call should exit non-zero if invoked with
     option --retcode-passthrough
     """
-    src = os.path.join(RUNTIME_VARS.BASE_FILES, "top.sls")
-    dst = os.path.join(RUNTIME_VARS.BASE_FILES, "top.sls.bak")
-    shutil.move(src, dst)
-    expected_comment = "No states found for this minion"
-    try:
-        ret = salt_call_cli.run("--retcode-passthrough", "state.highstate")
-    finally:
-        shutil.move(dst, src)
-    assert ret.exitcode != 0
-    assert expected_comment in ret.stdout
+    top_sls = """
+    base:
+      '*':
+        - core
+        """
+
+    with temp_state_file("top.sls", top_sls) as src:
```

Comment:
```suggestion
    with pytest.helpers.temp_file("top.sls", top_sls, base_env_state_tree_root_dir) as src:
```

Reply:
In fact, why go through the trouble of creating the topfile, if we're moving it to cause a `No states found...`, just don't create anything?